### PR TITLE
Restructure v9 and v10 examples

### PR DESCRIPTION
**Pull request by MikeMcC399**
This PR follows on from PR https://github.com/cypress-io/github-action/pull/725 "Update Cypress v10 tests to 12.3.0".

The label `v10` no longer accurately describes examples running under the current version of Cypress 12.x and so the label is `v10` is removed. Examples running under Cypress 9.7.0 (which is the last version using [Legacy Configuration](https://docs.cypress.io/guides/references/legacy-configuration)) now use the label `v9` consistently.

1. The contents of the directory [examples/v10](https://github.com/cypress-io/github-action/tree/master/examples/v10), which now contains tests using Cypress 12.3.0, are moved to [examples](https://github.com/cypress-io/github-action/tree/master/examples) and the empty directory [examples/v10](https://github.com/cypress-io/github-action/tree/master/examples/v10) is deleted.
2. The [example workflows](https://github.com/cypress-io/github-action/tree/master/.github/workflows) and the [README.md](https://github.com/cypress-io/github-action/blob/master/README.md) file are changed to take account of the new working directory location of the Cypress 12.x tests.
3. Job names and artefacts in the example workflows are changed so that for ones related to tests on Cypress 12.x any suffix `-v10` is removed. Job names and artefacts related to tests on Cypress 9.x use the suffix `-v9`.

### CODE REVIEW

**Comment by MikeMcC399** [comment]
@jaffrepaul 

Please review and merge if it meets your expectations. You wrote in https://github.com/cypress-io/github-action/pull/725#pullrequestreview-1248342381

>  I think we should just remove the v10 level in a separate PR to imply cypress(latest) with the additional v9 set.

Although this isn't a breaking change for the action, it is quite disruptive for PRs that I have in the pipeline, which will need to be reworked. All my other work is on hold until this PR is dealt with.

File: .github/workflows/example-basic-pnpm.yml
Comment (jaffrepaul) [inline .github/workflows/example-basic-pnpm.yml:8]: I suppose "and lower" doesn't accurately describe this. I think we can just edit the comment to read "Cypress v9" in all applicable areas to avoid any confusion. I don't mind "v10 and higher" since it conveys the larger change in post v9 and expands to the latest Cypress version.

Comment (MikeMcC399) [reply .github/workflows/example-basic-pnpm.yml:8]: @jaffrepaul 

Instead of changing to "Cypress v9" (knocking off "and lower"), I suggest instead to align with the documentation for [Cypress legacy configuration](https://docs.cypress.io/guides/references/legacy-configuration) which says "This guide is for Cypress 9 and below" and change the header title instead to

"Cypress v9 and below (using Legacy configuration)"

which reflects the fact that the v9 tests are representing all legacy configurations.

It's your call. I will change the title to what you think is best, either:
- "Cypress v9" or
- "Cypress v9 and below (using Legacy configuration)"

Let me know.

File: .github/workflows/example-custom-ci-build-id.yml
Comment (MikeMcC399) [reply .github/workflows/example-custom-ci-build-id.yml:60]: @jaffrepaul 

Testing from a fork is a bit tricky. I tested it in a private copy of the repository a few days ago. Unhappily I have now used up all my free GitHub minutes for the month of January now, so I can't repeat that test. 🙁 

To make doubly sure, I retested using a separate branch in my public fork [test/remove-v10-label](https://github.com/MikeMcC399/github-action/tree/test/remove-v10-label), using my own private project id.

Job [6785584499](https://github.com/MikeMcC399/github-action/actions/runs/3960869653/jobs/6785584499) shows it recording the smoke test and I confirmed that I could also view the test in the Cypress Cloud web UI.

Unfortunately it is not quite so easy to test a recording PR from a fork. I hope to improve on this when we get around to looking at the draft PR https://github.com/cypress-io/github-action/pull/694 at some later stage.

So, yes, I can confirm that things are working as expected.

**Comment by MikeMcC399** [comment]
The failure in `example-wait-on.yml` logged in https://github.com/cypress-io/github-action/actions/runs/3961225660/jobs/6786408657 was HTTP 429 "too many requests". 

I suspect this is a transient error only.

File: .github/workflows/example-custom-ci-build-id.yml
Comment (jaffrepaul) [reply .github/workflows/example-custom-ci-build-id.yml:60]: Understood, thank you.

**Comment by jaffrepaul** [comment]
@MikeMcC399 yeah I've seen some intermittent test failures in the pipeline today with some timeouts. I re ran the above test and it's looking good now. Something to keep an eye on.

**Review by jaffrepaul — APPROVED** [review]

**Pull request merged**
